config/stack.go

```diff
@@ -273,12 +273,14 @@ func processComponent(ctx context.Context, opts *options.TerragruntOptions, cmp
 		return errors.Errorf("failed to adjust source %s: %v", cmp.source, err)
 	}
 
-	dest := cmp.path
-	// if destination is not an absolute path, join with target directory
-	if !filepath.IsAbs(cmp.path) {
-		dest = filepath.Join(cmp.targetDir, cmp.path)
+	if filepath.IsAbs(cmp.path) {
+		return errors.Errorf("path %s must be relative", cmp.path)
 	}
 
+	// building destination path based on target directory
+	dest := cmp.path
+	dest = filepath.Join(cmp.targetDir, cmp.path)
```

Comment:
We should do another check here to see if the user is breaking out of the current working directory. check if the directory has a prefix of the directory where the `terragrunt.stack.hcl` file is.